nexus/mgs-updates/src/rot_updater.rs

```diff
@@ -203,23 +210,218 @@ impl SpComponentUpdateHelper for ReconfiguratorRotUpdater {
+                .try_all_serially(log, move |mgs_client| async move {
+                    mgs_client
+                        .sp_component_caboose_get(
+                            update.sp_type,
+                            update.slot_id,
+                            &SpComponent::ROT.to_string(),
+                            expected_active_slot.toggled().to_u16(),
+                        )
+                        .await
+                })
+                .await;
+            let found_version = match found_inactive_caboose_result {
+                Ok(version) => {
+                    FoundVersion::Version(version.into_inner().version)
+                }
+                Err(error) => {
+                    if error_means_caboose_is_invalid(&error) {
+                        FoundVersion::MissingVersion
+                    } else {
+                        return Err(PrecheckError::from(error));
+                    }
+                }
+            };
+            match (&expected_inactive_slot_version, &found_version) {
+                // expected garbage, found garbage
+                (
+                    ExpectedVersion::NoValidVersion,
+                    FoundVersion::MissingVersion,
+                ) => (),
+                // expected a specific version and found it
+                (
+                    ExpectedVersion::Version(artifact_version),
+                    FoundVersion::Version(found_version),
+                ) if artifact_version.to_string() == *found_version => (),
+                // anything else is a mismatch
+                (ExpectedVersion::NoValidVersion, FoundVersion::Version(_))
+                | (ExpectedVersion::Version(_), FoundVersion::MissingVersion)
+                | (ExpectedVersion::Version(_), FoundVersion::Version(_)) => {
+                    return Err(PrecheckError::WrongInactiveVersion {
+                        expected: expected_inactive_slot_version.clone(),
+                        found: found_version,
+                    });
+                }
+            };
+
+            let (
+                active, pending_persistent_boot_preference, transient_boot_preference
+            ) = match &state.rot {
+                RotState::V2 {
+                    active,
+                    pending_persistent_boot_preference,
+                    transient_boot_preference,
+                    ..
+                }
+                | RotState::V3 {
+                    active,
+                    pending_persistent_boot_preference,
+                    transient_boot_preference,
+                    ..
+                } => (
+                    active, pending_persistent_boot_preference, transient_boot_preference
+                ),
+                RotState::CommunicationFailed { message } => {
+                    return Err(PrecheckError::RotCommunicationFailed { message: message.to_string() })
+                },
+            };
```

Comment:
As far as I understand we are getting these values from boot info. From what I know, there is a chance these values may not reflect the real state of the component, as these are taken at boot time. It seems very unlikely these would change in production, but just want to double check it's safe to use them.

Reply:
Sorry, I don't understand this comment - maybe a question for @lzrd. What is "real state" vs "boot time" here?

Reply:
Spoke about this during the update watercooler. version, git commit and sign are only taken at boot time, but the active slot, pending_persistent_boot_preference and transient_boot_preference reflect the real state of the component